### PR TITLE
ARROW-12019: [Rust] [Parquet] Update README for 2.6.0 support

### PR DESCRIPTION
**Pull request by nevi-me**
Updates the README for 2.6.0 support, and updates the notes about ConvertedType & LogicalType

### CODE REVIEW

**Review by jorgecarleitao — APPROVED** [review]
LGTM 👍  Great progress!!

File: rust/parquet/README.md [hunk 1]
```diff
@@ -50,14 +50,18 @@ See [crate documentation](https://docs.rs/crate/parquet/4.0.0-SNAPSHOT) on avail
...
+This crate used `parquet::basic::LogicalType` to map to the `ConvertedType`, but this has been renamed to `parquet::basic::ConvertedType` from version 4.0 of this crate.
+
+The `ConvertedType` is deprecated in the format, but is still written
+to preserve backward compatibility.
+It is preferred that `LogicalType` is used, as it suports nanosecond
+precision timestamps without using the deprecated `Int96` Parquet type.
 
 ## Supported Parquet Version
-- Parquet-format 2.4.0
+- Parquet-format 2.6.0
```
Comment (alamb) [inline rust/parquet/README.md:64]: 🎉

**Review by alamb — APPROVED** [review]

File: rust/parquet/README.md [hunk 2]
```diff
@@ -50,14 +50,18 @@ See [crate documentation](https://docs.rs/crate/parquet/4.0.0-SNAPSHOT) on avail
...
-Parquet Format 2.4.0. The type previously called `LogicalType` was was
-renamed to `ConvertedType`.
+[`LogicalType`] to preserve existing behaviour in your code.
 
+Version 2.4.0 of the Parquet format introduced a `LogicalType` to replace the existing `ConvertedType`.
+This crate used `parquet::basic::LogicalType` to map to the `ConvertedType`, but this has been renamed to `parquet::basic::ConvertedType` from version 4.0 of this crate.
+
+The `ConvertedType` is deprecated in the format, but is still written
+to preserve backward compatibility.
+It is preferred that `LogicalType` is used, as it suports nanosecond
```
Comment (sunchao) [inline rust/parquet/README.md:60]: suports -> supports

**Review by sunchao — APPROVED** [review]
+1

File: rust/parquet/README.md (hunk 2 above)
Comment (alamb) [reply rust/parquet/README.md:60]:
```suggestion
It is preferred that `LogicalType` is used, as it supports nanosecond
```

**Pull request closed without merging**